a3p-integration/proposals/n:upgrade-next/psm.test.js

```diff
@@ -29,38 +27,19 @@ test.before(async t => {
   };
 });
 
-test.serial('simulate trade of IST and USDC', async t => {
+test.serial('check stats pre-swap', async t => {
```

Comment:
I won't block on this, but we've been abusing `test.serial`. Its only guarantee is "not serial" and we've been relying on it being sequential.

Consider factoring into one `test()` call. If the function is unwieldy you can use a driver model to abstract the steps. Or just put a log message for each stage.

Reply:
> I won't block on this, but we've been abusing `test.serial`. Its only guarantee is "not serial" and we've been relying on it being sequential.

You mean "not concurrent". I honestly didn't realize it didn't guarantee sequential.

Reply:
Yeah thanks for correcting my write-o.

https://github.com/avajs/ava/blob/main/docs/01-writing-tests.md#running-tests-serially

https://github.com/avajs/ava/discussions/3238